lib/delayed/psych_ext.rb

```diff
@@ -74,7 +74,11 @@ def visit_Psych_Nodes_Mapping(object) # rubocop:disable CyclomaticComplexity, Me
             raise Delayed::DeserializationError, "DataMapper::ObjectNotFoundError, class: #{klass} (#{error.message})"
           end
         else
-          super
+          begin
+            super
+          rescue Exception => error
```

Comment:
We could catch `ScriptError` instead; this is the superclass of NotImplementedError (the one that's thrown for this particular case) as well as `SyntaxError` and `LoadError`